protocol/message.go

```diff
@@ -216,6 +260,40 @@ func NewMonitoringProof(ap []*m.AuthenticationPath,
 	}, ReqSuccess
 }
 
+// NewObservedSTR creates the response message a CONIKS auditor
+// sends to a client, or a CONIKS directory sends to an auditor,
+// upon an AuditingRequest, and returns a Response containing an ObservedSTR struct.
+// auditlog.GetObservedSTR(), or directory.GetLatestSTR(), passes the signed
+// tree root for the directory's latest str.
+//
+// See auditlog.Audit() for details on the contents of the created
+// ObservedSTR.
+func NewObservedSTR(str *m.SignedTreeRoot) (*Response, ErrorCode) {
+	return &Response{
+		Error: ReqSuccess,
+		DirectoryResponse: &ObservedSTR{
+			STR: str,
+		},
+	}, ReqSuccess
+}
+
+// NewObservedSTRs creates the response message a CONIKS auditor
+// sends to a client upon an AuditingInEpochRequest,
+// and returns a Response containing an ObservedSTRs struct.
+// auditlog.AuditInEpoch() passes a list of signed tree roots
+// that the auditor observed for the requested range of epochs str.
+//
+// See auditlog.AuditInEpoch() for details on the contents of the created
+// ObservedSTRs.
+func NewObservedSTRs(str []*m.SignedTreeRoot) (*Response, ErrorCode) {
+	return &Response{
+		Error: ReqSuccess,
+		DirectoryResponse: &ObservedSTRs{
+			STR: str,
+		},
+	}, ReqSuccess
+}
+
 func (msg *Response) validate() error {
```

Comment:
Yes, I think this makes sense, especially if we're going to have some generic `HandleSTRs` function.